src/uct/rocm/ipc/rocm_ipc_ep.c

```diff
@@ -94,20 +99,46 @@ ucs_status_t uct_rocm_ipc_ep_zcopy(uct_ep_h tl_ep,
 
     remote_copy_addr = UCS_PTR_BYTE_OFFSET(remote_base_addr,
                                            remote_addr - key->address);
+
+    memset(&remote_agent, 0, sizeof(hsa_agent_t));
+    status = uct_rocm_base_get_ptr_info(remote_copy_addr, size, &tmp_base_ptr,
+					&tmp_base_size,	&remote_agent);
+    if (status != HSA_STATUS_SUCCESS) {
+	return UCS_ERR_INVALID_ADDR;
+    }
+
+    if (remote_agent.handle == 0) {
+	//No access to remote agent, e.g. because of limited visability of devices to
+	//this process. Using local_agent as a backup plan.
+	remote_agent = local_agent;
+    }
+    else {
```

Comment:
code style
pls check the output of "Code format" stage in the CI

Reply:
if the issue here is the location of the 'else' statement, I moved it up to follow the rest of the ucx code. I also saw that there are some tabs/spaces issues, I fixed them. Beyond it, I am working on compiling the llvm tool to run the clang-format manually (which takes however a while), since the output of the CI stage does not actually tell what is wrong with the formatting, it just says that it 'passed with issues'.

Reply:
@edgargabriel  pls see https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=43598&view=logs&j=cc064a77-22b5-56bf-ecc0-70b5fe764261&t=aedfd754-44a6-53e6-6843-8659d800fee2